tools/dds/dds-server/dds-device-broadcaster.cpp

```diff
@@ -261,29 +211,41 @@ bool tools::dds_server::send_device_info_msg( const librealsense::dds::topics::d
     return false;
 }
 
-librealsense::dds::topics::device_info dds_server::query_device_info( const rs2::device &rs2_dev ) const
+librealsense::dds::topics::device_info dds_device_broadcaster::query_device_info( const rs2::device &rs2_dev ) const
 {
     librealsense::dds::topics::device_info dev_info;
     dev_info.name = rs2_dev.get_info( RS2_CAMERA_INFO_NAME );
     dev_info.serial = rs2_dev.get_info( RS2_CAMERA_INFO_SERIAL_NUMBER );
     dev_info.product_line = rs2_dev.get_info( RS2_CAMERA_INFO_PRODUCT_LINE );
     dev_info.locked = (rs2_dev.get_info( RS2_CAMERA_INFO_CAMERA_LOCKED ) == "YES");
+
+    // Build device topic root path
+    dev_info.topic_root = get_topic_root(dev_info.name, dev_info.serial);
     return dev_info;
 }
 
-void dds_server::fill_device_msg( const librealsense::dds::topics::device_info & dev_info,
+void dds_device_broadcaster::fill_device_msg( const librealsense::dds::topics::device_info & dev_info,
                                   librealsense::dds::topics::raw::device_info & msg ) const
 {
     strcpy( msg.name().data(), dev_info.name.c_str() );
     strcpy( msg.serial_number().data(), dev_info.serial.c_str() );
     strcpy( msg.product_line().data(), dev_info.product_line.c_str() );
+    strcpy( msg.topic_root().data(), dev_info.topic_root.c_str() );
     msg.locked() = dev_info.locked;
 }
 
-dds_server::~dds_server()
+std::string tools::dds_device_broadcaster::get_topic_root( const std::string& name, const std::string& sn ) const
```

Comment:
I think maybe we should pass the whole `device_info` and let this function take what it needs

Reply:
👍